### PR TITLE
quiche: adjust test size of quic_http_integration_test

### LINKED ISSUE #15407: Flaky test: //test/extensions/quic_listeners/quiche/integration:quic_http_integration_test
https://dev.azure.com/cncf/4684fb3d-0389-4e0b-8251-221942316e06/_apis/build/builds/68704/logs/212

//test/extensions/quic_listeners/quiche/integration:quic_http_integration_test failed with a timeout on TSAN on my PR with unrelated changes.

### PR DESCRIPTION
**Pull request by danzh2010**
Commit Message: Increase the test size as it doesn't have enough time to finish under TSAN.

Risk Level: low
Testing: n/a
Fixes #15407

### CODE REVIEW

File: test/extensions/quic_listeners/quiche/integration/BUILD [hunk 1]
```diff
@@ -32,9 +32,10 @@ envoy_cc_test(
 
 envoy_cc_test(
     name = "quic_http_integration_test",
-    size = "medium",
+    size = "large",
```
Comment (alyssawilk) [inline test/extensions/quic_listeners/quiche/integration/BUILD:35]: oh goodness I hadn't noticed the tagging, this should definitely help.

File: test/extensions/quic_listeners/quiche/integration/BUILD [hunk 2]
```diff
@@ -32,9 +32,10 @@ envoy_cc_test(
...
     srcs = ["quic_http_integration_test.cc"],
     data = ["//test/config/integration/certs"],
+    shard_count = 12,
```
Comment (alyssawilk) [inline test/extensions/quic_listeners/quiche/integration/BUILD:38]: protocol_integration_test has only 5 shards - are 12 necessary?  runtime of 50-60s is pretty normal so as long as it doesn't take several minutes I think we're good.

**Review by alyssawilk — COMMENTED** [review]
Awesome, thanks for getting right on this!

File: test/extensions/quic_listeners/quiche/integration/BUILD (hunk 2 above)
Comment (danzh2010) [reply test/extensions/quic_listeners/quiche/integration/BUILD]: This tests runs a combination of {IPv4, IPv6} and {gQUIC with Quic Crypto, gQUIC with TLS, iQUIC with TLS}. So I reduced it to 6.

**Review by alyssawilk — APPROVED** [review]
cc @mattklein123 @htuch @lizan for merge, since verify examples is failing and I can't see this PR being a problem.

**Pull request merged**